### DISCUSSION TITLE
Glow-TTS and vocoder

### DISCUSSION THREAD

**Discussion by madey83** [idiap/coqui-ai-TTS#122] [Q&A] [answered]
Hellow All,

i could not find any information if Glow-TTS model requires vocoder model when i would like to training from scrach polish dataset model?

if yes, vocoder will uses the same dataset like for Glow-TTS?

Can i ask for advice?

**Comment by eginhard** [comment, marked as answer]
You don't necessarily have to train your own vocoder, you can use a pretrained one. If you're just running the Glow-TTS recipe it will only train the TTS model, the vocoder recipes are separate.

**Reply by madey83** [reply]
ok, so is there any advantage to have vocoder available during training model? And what is vocoder use for?

**Reply by madey83** [reply]
i have foud this https://github.com/jaywalnut310/glow-tts/blob/master/configs/base_blank.json which below statement:

This result was not included in the paper. Lately, we found that two modifications help to improve the synthesis quality of Glow-TTS.; 1) moving to a vocoder, [HiFi-GAN](https://arxiv.org/abs/2010.05646) to reduce noise, 2) putting a blank token between any two input tokens to improve pronunciation.

Does anyone try it?

**Reply by eginhard** [reply]
Many TTS models, including Glow, only convert from text to Mel spectrograms. The vocoder is needed to convert these into a speech signal then, so that you can actually listen to the outputs. Coqui TTS has implementations and pretrained models for different vocoders, including HiFi-GAN, so you can pick your favourite.

You can also train with such a blank token by setting `add_blank=True` in the config, see https://github.com/idiap/coqui-ai-TTS/blob/88de5c4f9be56b8e73a9ec002a8e04c48f12e00a/recipes/bel-alex73/train_glowtts.py#L61

**Reply by madey83** [reply]
sorry for too many basic questions but i'm starting with this topic and i try to understand what to do in order to get some good results...

So i can train glow model for polish language with HiFi-GAN vocoder pre-trained for other language?

**Reply by madey83** [reply]
thank you for all answers...

**Comment by madey83** [comment]
can this [https://github.com/idiap/coqui-ai-TTS/blob/88de5c4f9be56b8e73a9ec002a8e04c48f12e00a/recipes/bel-alex73/train_glowtts.py#L61](url) be used as is or should i modify it based on my RTX2060 GPU hardware limitation?

any sugesstion?
